homeassistant/components/media_player/intent.py

```diff
@@ -77,11 +115,14 @@ async def async_handle_states(
         match_preferences: intent.MatchTargetsPreferences | None = None,
     ) -> intent.IntentResponse:
         """Record last paused media players."""
-        hass = intent_obj.hass
-
         if match_result.is_match:
             # Save entity ids of paused media players
-            hass.data[DATA_LAST_PAUSED] = {s.entity_id for s in match_result.states}
+            self.last_paused.timestamp = time.time()
+            self.last_paused.entity_ids.clear()
+            self.last_paused.waiting_entity_ids.clear()
+            self.last_paused.waiting_entity_ids.update(
+                s.entity_id for s in match_result.states
+            )
```

Comment:
This feels hacky. Cna you make a setter function.

Reply:
Moved to an update function